### PR TITLE
fix(DB/Creature): Prince Sandoval is unattackable

### PR DESCRIPTION
**Pull request by locus313**
## CHANGES PROPOSED:
-  Update creature_template for Prince Sandoval
-  Update smart_scripts for Prince Sandoval

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3135
- Credits:  r0m1ntik94

## TESTS PERFORMED:
- Apply sql and restart server
- .tele icecrown
- .quest add 13213
- went through the quest chain 
- was able to kill Prince Sandoval at the end of the quest chain

## HOW TO TEST THE CHANGES:

- Apply sql and restart server
- .tele icecrown
- .quest add 13213
- progress through the quest chain 
- see if you can kill Prince Sandoval at the end of the quest chain

## KNOWN ISSUES AND TODO LIST:

- [x] fix invalid event flags

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1598238244990167657');
+
+UPDATE `creature_template` SET `faction` = 21 WHERE `entry` = 14688;
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 14688;
+INSERT INTO `smart_scripts` VALUES
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:5]: You are missing [...] (`entryorguid, [...]) VALUES

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1598238244990167657');
+
+UPDATE `creature_template` SET `faction` = 21 WHERE `entry` = 14688;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:3]: We need to verify that this change is correct and do not break other stuff

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 3]
```diff
@@ -7,22 +7,22 @@ INSERT INTO `smart_scripts` VALUES
...
+	(14688, 0, 13, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 11, 4335, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Cast \'Spawn Smoke (scale x2.00)\''),
+	(14688, 0, 14, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Say Line 0'),
 	(14688, 0, 15, 0, 1, 0, 100, 1, 14000, 14000, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 21, 100, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Start Attacking'),
-	(14688, 0, 16, 0, 5, 0, 100, 512, 0, 0, 0, 0, 0, 45, 20, 20, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Killed Unit - Set Data 20 20'),
-	(14688, 0, 17, 0, 6, 0, 100, 512, 0, 0, 1, 0, 0, 45, 6, 6, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Died - Set Data 6 6'),
-	(14688, 0, 18, 0, 1, 0, 100, 513, 30000, 30000, 0, 0, 0, 45, 19, 19, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Set Data 19 19'),
+	(14688, 0, 16, 0, 5, 0, 100, 0, 0, 0, 0, 0, 0, 45, 20, 20, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Killed Unit - Set Data 20 20'),
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:22]: Eventflag should be 1

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 4]
```diff
@@ -7,22 +7,22 @@ INSERT INTO `smart_scripts` VALUES
...
-	(14688, 0, 9, 10, 61, 0, 100, 512, 0, 0, 0, 0, 0, 28, 61144, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Remove Aura \'Fire Shield\''),
-	(14688, 0, 10, 11, 61, 0, 100, 512, 0, 0, 0, 0, 0, 28, 61145, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Remove Aura \'Ember Shower\''),
-	(14688, 0, 11, 0, 61, 0, 100, 512, 0, 0, 0, 0, 0, 103, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Set Rooted Off'),
-	(14688, 0, 12, 0, 54, 0, 100, 512, 0, 0, 0, 0, 0, 80, 1468800, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Run Script'),
-	(14688, 0, 13, 0, 54, 0, 100, 512, 0, 0, 0, 0, 0, 11, 4335, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Cast \'Spawn Smoke (scale x2.00)\''),
-	(14688, 0, 14, 0, 54, 0, 100, 513, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Say Line 0'),
+	(14688, 0, 4, 5, 61, 0, 100, 0, 0, 0, 0, 0, 0, 21, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Disable Combat Movement'),
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:10]: Eventflag should be 1

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql (hunk 3 above)
Comment (locus313) [reply data/sql/updates/pending_db_world/rev_1598238244990167657.sql]: updated the eventflag for these lines

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql (hunk 4 above)
Comment (locus313) [reply data/sql/updates/pending_db_world/rev_1598238244990167657.sql]: updated the eventflag for these lines

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 5]
```diff
@@ -7,21 +7,21 @@ INSERT INTO `smart_scripts` VALUES
...
 	(14688, 0, 15, 0, 1, 0, 100, 1, 14000, 14000, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 21, 100, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Start Attacking'),
-	(14688, 0, 16, 0, 5, 0, 100, 0, 0, 0, 0, 0, 0, 45, 20, 20, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Killed Unit - Set Data 20 20'),
-	(14688, 0, 17, 0, 6, 0, 100, 0, 0, 0, 1, 0, 0, 45, 6, 6, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Died - Set Data 6 6'),
-	(14688, 0, 18, 0, 1, 0, 100, 0, 30000, 30000, 0, 0, 0, 45, 19, 19, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Set Data 19 19'),
+	(14688, 0, 16, 0, 5, 0, 100, 1, 0, 0, 0, 0, 0, 45, 20, 20, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Killed Unit - Set Data 20 20'),
+	(14688, 0, 17, 0, 6, 0, 100, 1, 0, 0, 1, 0, 0, 45, 6, 6, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Died - Set Data 6 6'),
+	(14688, 0, 18, 0, 1, 0, 100, 1, 30000, 30000, 0, 0, 0, 45, 19, 19, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Set Data 19 19'),
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:24]:
```suggestion
	(14688, 0, 16, 0, 5, 0, 100, 0, 0, 0, 0, 0, 0, 45, 20, 20, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Killed Unit - Set Data 20 20'),
	(14688, 0, 17, 0, 6, 0, 100, 0, 0, 0, 1, 0, 0, 45, 6, 6, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Died - Set Data 6 6'),
	(14688, 0, 18, 0, 1, 0, 100, 1, 30000, 30000, 0, 0, 0, 45, 19, 19, 0, 0, 0, 0, 11, 31135, 100, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - Out of Combat - Set Data 19 19'),
```

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql [hunk 6]
```diff
@@ -7,21 +7,21 @@ INSERT INTO `smart_scripts` VALUES
...
 	(14688, 0, 10, 11, 61, 0, 100, 0, 0, 0, 0, 0, 0, 28, 61145, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Remove Aura \'Ember Shower\''),
 	(14688, 0, 11, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 103, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - In Combat - Set Rooted Off'),
-	(14688, 0, 12, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 80, 1468800, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Run Script'),
-	(14688, 0, 13, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 11, 4335, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Cast \'Spawn Smoke (scale x2.00)\''),
-	(14688, 0, 14, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Say Line 0'),
+	(14688, 0, 12, 0, 54, 0, 100, 1, 0, 0, 0, 0, 0, 80, 1468800, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Run Script'),
+	(14688, 0, 13, 0, 54, 0, 100, 1, 0, 0, 0, 0, 0, 11, 4335, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Cast \'Spawn Smoke (scale x2.00)\''),
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1598238244990167657.sql:19]:
```suggestion
	(14688, 0, 12, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 80, 1468800, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Run Script'),
	(14688, 0, 13, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 11, 4335, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Prince Sandoval - On Just Summoned - Cast \'Spawn Smoke (scale x2.00)\''),
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]
My bad, I checked wrong.

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql (hunk 1 above)
Comment (locus313) [reply data/sql/updates/pending_db_world/rev_1598238244990167657.sql]: updated the insert

**Comment by locus313** [comment]
> My bad, I checked wrong.

no worries, thanks for your help.

**Comment by locus313** [comment]
@Kitzunu I just tested the changes, now none of the boss in the quest chain attack you. :(

**Comment by Kitzunu** [comment]
did they do before my suggestion?

**Comment by locus313** [comment]
yeah all boss worked with eventflag 0

**Comment by Kitzunu** [comment]
change to 0

**Comment by locus313** [comment]
reverted back to 0 and retested. all boss are working again.

File: data/sql/updates/pending_db_world/rev_1598238244990167657.sql (hunk 2 above)
Comment (locus313) [reply data/sql/updates/pending_db_world/rev_1598238244990167657.sql:3]: double checked the faction here:
https://wowgaming.altervista.org/aowow/?npc=14688

Looked up the factiontemplate:
https://www.azerothcore.org/wiki/FactionTemplate

21 looks to be correct. 
Anything else i should check?

**Comment by felwyn** [comment]
Confirmed. Is working. But the whole quest series would probably need some Quality fixes

**Pull request merged**